gapii/cc/spy.cpp

```diff
@@ -495,6 +496,8 @@ bool Spy::checkEndTrace() {
     // mConnection->write(err.data(), err.size());
     auto msg = protocol::createHeader(protocol::MessageType::kEndTrace);
     mConnection->write(msg.data(), msg.size());
+    // allow some time for the message to arrive
+    std::this_thread::sleep_for(std::chrono::milliseconds(200));
```

Comment:
Can't we have a call to some kind of `flush()` instead?

Reply:
There is no "flush" functionality for sockets. It would be possible to enable the linger option of the socket, but this would require some time to implement and test in core\cc\socket_connection.cpp

Reply:
Just tried the linger option for the socket, but it didn't fix the problem, so we have to keep the sleep or change the protocol to send an acknowledge or similar.